internal/gcs/bridge_test.go

```diff
@@ -0,0 +1,130 @@
+package gcs
+
+import (
+	"context"
+	"encoding/binary"
+	"io"
+	"strings"
+	"testing"
+	"time"
+
+	"github.com/sirupsen/logrus"
+)
+
+type stitched struct {
+	io.ReadCloser
+	io.WriteCloser
+}
+
+func (s *stitched) Close() error {
+	s.ReadCloser.Close()
+	s.WriteCloser.Close()
+	return nil
+}
+
+func pipeConn() (*stitched, *stitched) {
+	r1, w1 := io.Pipe()
+	r2, w2 := io.Pipe()
+	return &stitched{r1, w2}, &stitched{r2, w1}
+}
+
+func reflector(t *testing.T, rw io.ReadWriteCloser, delay time.Duration) {
+	defer rw.Close()
+	for {
+		id, typ, msg, err := readMessage(rw)
+		if err != nil {
+			if err != io.EOF {
+				t.Error(err)
+			}
+			return
+		}
+		time.Sleep(delay)
```

Comment:
Can you just add a comment here that says `// delay is used to test timeout cases for responses and otherwise will be 0`. It didn't make any sense to me why we would want a delay in the tests at first.